crates/server/src/store/blocklist/config.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2015-2022 Benjamin Fry <benjaminfry@me.com>
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// https://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// https://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+use serde::Deserialize;
+
+/// Consult action enum.  Controls how consult lookups are handled.
+#[derive(Clone, Copy, Debug, Deserialize, Eq, PartialEq)]
+pub enum BlocklistConsultAction {
+    Disabled,
+    Enforce,
+    Log,
+}
+
+/// Configuration for file based zones
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq)]
+#[serde(deny_unknown_fields)]
+pub struct BlocklistConfig {
+    /// Support wildcards?  If set to true, block list entries containing asterisks will be expanded
+    /// to match queries.
+    #[serde(default = "wildcard_match_default")]
+    pub wildcard_match: bool,
+
+    /// Minimum wildcard depth.  Defaults to 2.  Any wildcard entries without at least this many
+    /// static elements will not be expanded (e.g., *.com has a depth of 1; *.example.com has a
+    /// depth of two.) This is meant as a safeguard against an errant block list entry, such as *
+    /// or *.com that might block many more hosts than intended.
+    #[serde(default = "min_wildcard_depth_default")]
+    pub min_wildcard_depth: u8,
+
+    /// Block lists to load.  These should be specified as relative (to the server zone directory)
+    /// paths in the config file.
+    pub lists: Vec<String>,
+
+    /// Sinkhole IP. This is the IP that is returned when a blocklist entry is matched.  It defaults
+    /// to 0.0.0.0.
+    #[serde(default = "sinkhole_ip_default")]
+    pub sinkhole_ip: String,
+
+    /// The consult action controls how the blocklist handles queries where another authority has
+    /// already provided an answer.  By default, it ignores any such queries ("Disabled",) however
+    /// it can be configured to log blocklist matches for those queries ("Log",) or can be
+    /// configured to overwrite the previous responses ("Enforce".)
+    #[serde(default = "consult_default")]
+    pub consult_action: BlocklistConsultAction,
+}
+
+impl BlocklistConfig {
+    /// the set of block lists which should be loaded
+    pub fn get_block_lists(&self) -> &Vec<String> {
+        &self.lists
+    }
+}
+
+fn wildcard_match_default() -> bool {
+    true
+}
+fn min_wildcard_depth_default() -> u8 {
+    2
+}
+fn sinkhole_ip_default() -> String {
+    String::from("0.0.0.0")
+}
+fn consult_default() -> BlocklistConsultAction {
+    BlocklistConsultAction::Disabled
+}
```

Comment:
I think we can write these directly in the attributes? The indirection here seems unhelpful.

Reply:
I may be missing something, but the only two ways I know of to provide default values for serde to use in deserialization are 1) to write functions like these, or 2) provide a Default implementation for the struct.  I changed this to have a Default implementation.

Reply:
Yeah, I think you're right -- sorry about that.